piff/gp_interp_2pcf.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: gp_interp_2pcf
+"""
+
+import numpy as np
+import treecorr
+import copy
+
+from sklearn.gaussian_process.kernels import Kernel
+import scipy.optimize as op
+from scipy.linalg import cholesky, cho_solve
+
+from .interp import Interp
+from .star import Star, StarFit
+
+
+class GPInterp2pcf(Interp):
+    """
+    An interpolator that uses two-point correlation function and gaussian process to interpolate a single surface.
+
+    :param keys:        A list of star attributes to interpolate from
+    :param kernel:      A string that can be `eval`ed to make a
+                        sklearn.gaussian_process.kernels.Kernel object.  The reprs of
+                        sklearn.gaussian_process.kernels will work, as well as the repr of a
+                        custom piff VonKarman object.  [default: 'RBF()']
```

Comment:
I did few test and if you write just ’RBF’ it will execute, but without initializing the kernel, which is a source of error if you are doing something like that right now. However if you write ‘2 *RBF’ it will not worked and you will get an error message. I don’t think it should work like that. I mean I guess this is better to feed the gp with initial guess of hyperparameter like it is done during all the test and so write you kernel with something like ‘RBF(2)’. I added a TypeError within the init in order to not allowed something like ‘RBF’ or ‘VonKarman’ alone.